dlp/snippets/deid/deid_table_condition_masking.go

```diff
@@ -0,0 +1,166 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deid
+
+// [START dlp_deidentify_table_condition_masking]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyTableMaskingCondition de-identifies the table data using masking
+// and conditional logic
+func deidentifyTableMaskingCondition(w io.Writer, projectID string) error {
+	// projectId := "your-project-id"
+
+	row1 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "22"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "Jane Austen"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "21"}},
+		},
+	}
+
+	row2 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "55"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "Mark Twain"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "75"}},
+		},
+	}
+
+	row3 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "101"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "Charles Dickens"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "95"}},
+		},
+	}
+
+	table := &dlppb.Table{
+		Headers: []*dlppb.FieldId{
+			{Name: "AGE"},
+			{Name: "PATIENT"},
+			{Name: "HAPPINESS SCORE"},
+		},
+		Rows: []*dlppb.Table_Row{
+			{Values: row1.Values},
+			{Values: row2.Values},
+			{Values: row3.Values},
+		},
+	}
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify what content you want the service to de-identify.
+	contentItem := &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_Table{
+			Table: table,
+		},
+	}
+
+	// Specify how the content should be de-identified.
+	charMaskConfig := &dlppb.CharacterMaskConfig{
+		MaskingCharacter: "*",
+	}
+	primitiveTransformation := &dlppb.PrimitiveTransformation_CharacterMaskConfig{
+		CharacterMaskConfig: charMaskConfig,
+	}
+
+	// Specify field to be de-identified.
+	fieldId := &dlppb.FieldId{
+		Name: "HAPPINESS SCORE",
+	}
+
+	// Specify when the above field should be de-identified.
+	condition := &dlppb.RecordCondition{
+		// Apply the condition to records
+		Expressions: &dlppb.RecordCondition_Expressions{
+			Type: &dlppb.RecordCondition_Expressions_Conditions{
+				Conditions: &dlppb.RecordCondition_Conditions{
+					Conditions: []*dlppb.RecordCondition_Condition{
+						{
+							Field:    fieldId,
+							Operator: dlppb.RelationalOperator_GREATER_THAN,
+							Value: &dlppb.Value{
+								Type: &dlppb.Value_IntegerValue{
```

Comment:
nit: nesting is a bit deep here. You might create this above--especially if you're working with protobuf.Value types.

Reply:
done.